contracts/base/strategies/multi/pipes/WrappingPipe.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: ISC
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./Pipe.sol";
+import "../../../../third_party/uniswap/IWETH.sol";
+
+/// @title Wrapping Pipe Contract
+/// @author bogdoslav
+contract WrappingPipe is Pipe {
+    /// @dev creates context
+    function create(address WETH) public pure returns (bytes memory){
+        return abi.encode(WETH);
+    }
+
+    /// @dev decodes context
+    function context(bytes memory c) internal pure returns (address WETH) {
+      (WETH) = abi.decode(c, (address));
+    }
+
+    /// @dev function for investing, deposits, entering, borrowing
+    function _put(bytes memory c, uint256 amount)
+    override public returns (uint256 output) {
```

Comment:
why this functions are public?

Reply:
payable?

Reply:
probably comment is wrong and it is fine to be public, please describe what this function do in comment

Reply:
changed to onlyOwner (owner = strategy contract)

Reply:
delegate calls removed from it, added onlyOwner